packages/rollup-core/src/app/data/data-service.ts

```diff
@@ -393,32 +393,37 @@ export class DefaultDataService implements DataService {
    * @inheritDoc
    */
   public async tryBuildCanonicalChainBatchNotPresentOnL1(
-    minBatchSize: number,
-    maxBatchSize: number
+    minBatchCalldataBytes: number,
+    maxBatchCalldataBytes: number
   ): Promise<number> {
     // TODO: ADD SOME SIZE LIMIT
     const txRes = await this.rdb.select(
```

Comment:
Verifying that I understand this change: calldata is stored in the db with the `0x` prefix, so `LENGTH(calldata)-2` accounts for the prefix.  The division by 2 here: `(LENGTH(calldata)-2) / 2` is converting from string size to byte size. Is there any chance that an odd size of calldata could end up in the database? I saw that ethers sometimes uses `0x0` as 0, the chainId is returned as a single character following `0x`. That is a different codepath but noting it here just in case. Empty calldata should be `0x`, if `LENGTH(calldata)` is 0 then there would be a bug here.

The `SUM` statement will add the calldata size (in bytes) of the 2 earliest transactions (`ORDER BY block_timestamp ASC`)

Reply:
> calldata is stored in the db with the 0x prefix, so LENGTH(calldata)-2 accounts for the prefix
Yep!

> The division by 2 here: (LENGTH(calldata)-2) / 2 is converting from string size to byte size.
Yep!

> Is there any chance that an odd size of calldata could end up in the database? I saw that ethers sometimes uses 0x0 as 0, the chainId is returned as a single character following 0x.
Yes and no. Yes, bad data can get into the DB (at which point we have way bigger problems -- we commit fraud), but no, calldata is always bytes, and bytes are always an even number of hex chars. Shortcuts used in JSON representation of some fields should not apply here.

> Empty calldata should be 0x, if LENGTH(calldata) is 0 then there would be a bug here.
Even empty calldata should be `0x` in the DB, but I'll make it `MAX(LENGTH(calldata)-2, 0)` just in case.

> The SUM statement will add the calldata size (in bytes) of the 2 earliest transactions (ORDER BY block_timestamp ASC)
This is the block timestamp of our L2 node, which we update infrequently, so there will be many txs with the same block timestamp. Two txs with different timestamps cannot be in the same rollup batch, though. So if this query returns 2 results, we know the first batch cannot get any bigger and _must_ be submitted. Regardless, this query is summing up all of the calldata of all of the txs that could possibly be in each batch (divided by block timestamp).